### PR TITLE
navigate to chat user's profile 

### LINKED ISSUE #4279: Can't navigate to user's profile
### User Story

As a user, I want to check user profile by clicking his user pic

### Description

*Type*: Feature

*Summary*: I want to navigate to user profile from chats or contacts list by clicking his user picture (any other use cases welcome, we don't have it designed, for now, so this is my personal assumption)

#### Expected behavior
Navigate to user profile:

1. Home - Appropriate chat - click user image
2. Contacts list - click user image

### Additional Information
* Operating System: MacOS

### PR DESCRIPTION
**Pull request by rcullito**
Fixes #4279

### CODE REVIEW

**Comment by rcullito** [comment]
A known issue is that this PR does not exactly fit the design spec here: zpl.io/V4J4NX4, mainly as we have not yet implemented dropdowns in RN for desktop. 

This PR addresses the issue of being able to navigate to the chat user's profile in the first place. Which can be done by clicking "View profile" in the upper right hand portion of the chat screen.

File: src/status_im/chat/events.cljs [hunk 1]
```diff
@@ -400,13 +400,6 @@
                               (navigate-to-chat random-id {})
                               (transport.message/send (group-chat/GroupAdminUpdate. chat-name selected-contacts) random-id)))))
 
-(handlers/register-handler-fx
```
Comment (cammellos) [inline src/status_im/chat/events.cljs]: Looks like this is actually still used, why did you remove it?

Comment (rcullito) [reply src/status_im/chat/events.cljs]: @cammellos thanks for spotting. I think initially I didn't think this exact handler would be able to be reused for desktop. But an error on my part to remove it entirely. Have since added it back in.

File: src/status_im/ui/components/desktop/events.cljs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  (:require [status-im.utils.handlers :as handlers]
+            [status-im.utils.handlers-macro :as handlers-macro]))
+
+(defn set-render-user [user {:keys [db]}]
+  "sets the user which should be visible on the desktop tab: profile"
+  (when user
+    {:db (assoc db :desktop/render-user user)}))
+
+(handlers/register-handler-fx
+ :show-desktop-tab
```
Comment (cammellos) [inline src/status_im/ui/components/desktop/events.cljs]: How does the logic exactly works?

Why can't we use `current-account` directly instead of storing it in the database under a different key (`desktop/render-user`)?

Comment (rcullito) [reply src/status_im/ui/components/desktop/events.cljs]: The goal was to be able to re-use the same tab for displaying either the current user's information or the profile of the user the "current user" is chatting with. 

Previously the user profile tab subscribed to the current account, and I introduced this new key so that it could be more flexible. `desktop/render-user` can take on a value of either the current account or the person who is being chatted with. The different values are set based on whether the app user clicks on their own profile or someone else's. 

Ideally it would be nice to pass the account as an argument to the tab profile component, but given how it is currently structured, that struck me as a larger refactor beyond the scope of this change. Happy to make that as part of this or another PR if that seems more sensible in the long run.

Comment (cammellos) [reply src/status_im/ui/components/desktop/events.cljs]: @rcullito I see, makes sense, I would recommend to move away from denormalising and do as you proposed, as otherwise the view will not re-render if the subscription is cached and a contact update is received, unless you also update the rendered-user, which is probably something we want to avoid, and try to keep state normalized unless necessary.

This PR another PR is all fine by me :)

File: src/status_im/ui/screens/desktop/main/chat/views.cljs [hunk 3]
```diff
@@ -17,21 +17,25 @@
...
           [icons/icon :icons/public-chat])
         (when (and group-chat (not public?))
           [icons/icon :icons/group-chat])
-        [react/text {:style styles/toolbar-chat-name}
-         chat-name]]
+                     [react/text {:style styles/toolbar-chat-name}
+         chat-name]
+                     (when (and (not group-chat) (not public?))
+                       [react/text {:on-press #(re-frame/dispatch [:show-desktop-tab :profile contact])}
+                       "View profile"])]
```
Comment (cammellos) [inline src/status_im/ui/screens/desktop/main/chat/views.cljs]: i18n

**Review by cammellos — APPROVED** [review]

File: src/status_im/ui/components/desktop/events.cljs (hunk 2 above)
Comment (rcullito) [reply src/status_im/ui/components/desktop/events.cljs]: @cammellos Thanks for the review! Right, good point about the benefits of re-rendering. And sounds good, will aim to "re-normalize" in this PR then before it proceeds too much further. :+1:

**Comment by rcullito** [comment]
@EugeOrtiz your design feedback on this would be greatly appreciated in addition to our usual QA. While we currently don't have modal support, I had to make a decision about where to place the chat user's profile, and opted to re-use the component that we are already using for the logged in user's profile. Let me know if you'd like to talk any more about this!

**Comment by churik** [comment]
Please, make sure that it is possible to build project on Jenkins.|
For now: 
https://github.com/status-im/status-react/pull/4893 cannot be built.
```
[MacOS parallel build stream] [31mWARNING[0m: Use of undeclared Var status-im.ui.screens.desktop.main.tabs.profile.views/profile at line 40 /Users/jenkins/workspace/sktop_pull-requests_PR-4893-OOJZB3WLBEPCUMJHRDGGECHNDWNBGRUFDQBP4NTHURV5TOWSMEWQ/src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs
[MacOS parallel build stream] Error encountered performing task 'cljsbuild' with profile(s): 'prod'
[MacOS parallel build stream] Subprocess failed
```
https://jenkins.status.im/job/status-react/job/desktop/job/pull-requests/view/change-requests/job/PR-4893/28/console

<blockquote><div><strong><a href="https://jenkins.status.im/job/status-react/job/desktop/job/pull-requests/view/change-requests/job/PR-4893/28/console">status-react » desktop » pull-requests » PR-4893 #28 Console [Jenkins]</a></strong></div></blockquote>

File: src/status_im/ui/components/desktop/tabs.cljs [hunk 4]
```diff
@@ -38,9 +39,7 @@
 (defn tab [index content view-id active?]
   [react/touchable-highlight {:style    (merge tabs.styles/tab-container {:flex 1})
                               :disabled active?
-                              :on-press #(do
-                                           (re-frame/dispatch [:navigate-to :home])
```
Comment (rcullito) [inline src/status_im/ui/components/desktop/tabs.cljs]: @siphiuel think this line is removed from my merge with some of your code. lmk if my misunderstanding is off here and we actually do need this for proper navigation!

**Comment by rcullito** [comment]
Hey guys. Planning on adding functionality for "Add to contact" in a separate PR as it feels like that's a separate bit of functionality than just navigating to the user's profile. 

For testing purposes. The functionality contained in this one is seen by clicking "View Profile" in the upper right hand corner when chatting with someone. Then once their profile is open, it should display, image, name, contact code, and option to either return to chat or add contact. All of this will be there with the above note that "add contact" isn't wired up yet.

**Review by maxhora — APPROVED** [review]

File: src/status_im/ui/screens/desktop/main/chat/styles.cljs [hunk 5]
```diff
@@ -96,4 +94,38 @@
...
 (def chat-view
   {:flex             1
-   :background-color :white})
\ No newline at end of file
+   :background-color :white})
+
+(def contact-card-text
+  {:color     colors/blue
+   :font-size 14})
+
```
Comment (cammellos) [inline src/status_im/ui/screens/desktop/main/chat/styles.cljs]: extra line

File: src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs [hunk 6]
```diff
@@ -32,6 +32,7 @@
        (i18n/label :sharing-copied-to-clipboard)]]
      [react/view {:style styles/tooltip-triangle}]]))
 
+
```
Comment (cammellos) [inline src/status_im/ui/screens/desktop/main/tabs/profile/views.cljs]: extra line

File: src/status_im/ui/screens/desktop/main/views.cljs [hunk 7]
```diff
@@ -16,21 +16,22 @@
...
 (views/defview main-view []
   (views/letsubs [view-id [:get :view-id]]
-    (let [component (case view-id
-                      :chat        chat.views/chat-view
-                      :new-contact add-new.views/new-contact
-                      :qr-code     profile.views/qr-code
-                      status-view)]
-      [react/view {:style {:flex 1}}
-       [component]])))
+                 (let [component (case view-id
```
Comment (cammellos) [inline src/status_im/ui/screens/desktop/main/views.cljs]: indentation

**Comment by rcullito** [comment]
@cammellos thanks for spotting those!

File: src/status_im/ui/components/desktop/tabs.cljs (hunk 4 above)
Comment (vitvly) [reply src/status_im/ui/components/desktop/tabs.cljs]: We don't want to persist right pane contents when navigating to a different tab. This was a problem with QR code PR, so I've added this event dispatch here. It resets right pane to a default view.

**Review by vitvly — APPROVED** [review]

File: src/status_im/ui/components/desktop/tabs.cljs (hunk 4 above)
Comment (rcullito) [reply src/status_im/ui/components/desktop/tabs.cljs]: Ah ok, thank you clarifying this. And yes that makes total sense. Will add it back in. :+1:

File: src/status_im/ui/screens/desktop/main/chat/views.cljs [hunk 8]
```diff
@@ -13,30 +13,36 @@
...
-                     (if (string/blank? name)
-                       (gfycat.core/generate-gfy public-key)
-                       (or name
-                           (i18n/label :t/chat-name))))]
-      [react/view {:style styles/toolbar-chat-view}
-       [react/view {:style {:flex-direction :row}}
+  (views/letsubs
+   [{:keys [chat-id name public-key public? group-chat]} [:get-current-chat]
+    {:keys [pending?] :as contact}                       [:get-current-chat-contact]]
+   (let [chat-name (str
```
Comment (vkjr) [inline src/status_im/ui/screens/desktop/main/chat/views.cljs]: I would suggest extracting this to a separate function. That will make the code more readable and maybe it is not the only place where we can reuse it.

**Review by vkjr — APPROVED** [review]

File: src/status_im/ui/screens/desktop/main/chat/views.cljs (hunk 8 above)
Comment (cammellos) [reply src/status_im/ui/screens/desktop/main/chat/views.cljs]: @vkjr good shout, actually there's already a subscription for that, `get-current-chat-name`, and should be used instead, reason being `:name` in chat is denormalized (needs to change everytime the contact updates her profile, therefore we are moving away from it)

Comment (rcullito) [reply src/status_im/ui/screens/desktop/main/chat/views.cljs]: good points! thanks both. will push a follow up change addressing this.

**Comment by churik** [comment]
1) Not possible to navigate to user profile from Public chat (clicking on user picture)
2)  Icons are too small and different from the design
icons
3) If User profile picture is white,edges are not visible (expected as grey lines, like in Mobile)
phpto
4) No padding (should be 32 px according to design)
padd
5) Contact code is not selectable for copying.
6) When User is in the Contact list, "In contacts" icon should be displayed. I believe even you are not implemented "Add to contact" functionality here, correct item have to be shown for already added user.
incont

**Comment by rcullito** [comment]
@churik thanks for your feedback! I've addressed most of your points, as well as those of @vkjr and @cammellos 

From our discussion last Friday, does it work if I make a separate PR for the remaining two items? This page re-uses a profile component from the other desktop tab, and I'm guessing viewing a profile of a user who is not already engaged in a chat with the current user will be a bit different in nature from what is in place currently. Still agreeing it needs to be done, just would like to tackle separately. Let me know if the changes checked off below seem valid. Thanks!

### Items to Address
- [x] remove chat-name code and use subscription 
- [x] update icons in chat profile (add is just a plus icon, for some reason other is not displaying well on desktop)
- [x] 32px padding for overall chat profile page
- [x] selectable contact code
- [x] dispay "in contacts" if the user is already in contacts
- [x] actually get "add to contacts" working from chat profile and have UI update accordingly 
- [ ] navigate to profile from public chat
- [ ] user profile photo gray border

**Comment by churik** [comment]
7) Conflict with latest `desktop` branch (output from Jenkins is [here](https://jenkins.status.im/job/status-react/job/desktop/job/pull-requests/view/change-requests/job/PR-4893/56/console))
8) Expected, that contact code in Profile should be like selectable text (look to https://github.com/status-im/status-react/issues/5103 for details).
For now, it is not obvious that user copied a code if he just clicked on it - no messages, no warning, and text is not selecatable.

For item "Navigate to Profile from Public chat" - yes we can create separate PR, if you feel it more comfortable.
For grey borders I'll create a separate issue - this case is not handled through all app.

**Comment by rcullito** [comment]
@churik thanks for those! k. 7 + 8 have been addressed when you have a chance to take another look at this. 

I'll open up another PR for navigating to profile from a public chat.

**Comment by churik** [comment]
A separate issue for navigating to user profile from Public chat: 
https://github.com/status-im/status-react/issues/5184
The issue with transfer profile to popup: 
https://github.com/status-im/status-react/issues/5186

**Comment by rcullito** [comment]
@churik great, thanks for testing and those separate issues. merging now :+1:
